### PR TITLE
Fix LightGBM test_examples runtime & cleanup resulting errors

### PR DESCRIPTION
**Pull request by ahuber21**
Cleanup of run_examples.py and test_examples.py

### CODE REVIEW

**Comment by ahuber21** [comment]
Updated CI: http://intel-ci.intel.com/ee7f073e-81da-f106-b4a0-a4bf010d0e2e

**Comment by ahuber21** [comment]
http://intel-ci.intel.com/ee7f1dde-e480-f15a-9024-a4bf010d0e2e

File: tests/test_examples.py [hunk 1]
```diff
@@ -63,7 +61,9 @@ def check_libraries(rule):
 
 
 # function reading file and returning numpy array
-def np_read_csv(f, c=None, s=0, n=np.iinfo(np.int64).max, t=np.float64):
+def np_read_csv(f, c=None, s=0, n=np.iinfo(np.int64).max, t=np.float64, **kwargs):
+    n = kwargs.get("nrows") or n
+    print(f"np_read_csv: {n=}")
```
Comment (ethanglaser) [inline tests/test_examples.py:66]:
```suggestion
```
if its for debugging

**Review by ethanglaser — APPROVED** [review]
LGTM and removes problematic CI examples...any other opinions on examples refactoring?

**Comment by ahuber21** [comment]
There is no loss in removing the examples for now, as they never ran previously due to missing dependencies. So the coverage is unaffected. Of course, the original intent was always to run examples as part of regular tests and we need to restore that. Separate issue though.

**Pull request merged**